classes/class09/README.md

```diff
@@ -1 +1,129 @@
-#Class 9
+#Class 9 - Application Design and Diagramming
+
+As we begin to develop larger and more complex applications, we'll want to start being methodical about how we design and plan them. 
+Today we will talk about strategies for planning and documenting large applications to make developing and debugging as simple as possible.
+
+##Planning
+
+###Specifications
+Writing a specification - or spec - for your application before you write any code can help not only you and your whole team understand the context for your application, but also serve as a road map for development.
+As our applications grow in size and complexity, having a document to refer to can keep you on track with developing features and managing complexity.
+
+As always, we defer to Wikipedia for an example of what a spec should contain such that it conforms to the *IEEE Standard for software specifications*.
+Yes that is actually a thing.
+
+- Introduction
+ - Purpose
+ - Definitions
+ - System overview
+ - References
+- Overall description
+ - Product perspective
+   - System Interfaces
+   - User Interfaces
+   - Hardware interfaces
+   - Software interfaces
+   - Communication Interfaces
+   - Memory Constraints
+   - Operations
+   - Site Adaptation Requirements
+ - Product functions
+ - User characteristics
+ - Constraints, assumptions and dependencies
+- Specific requirements
+ - External interface requirements
+ - Functional requirements
+ - Performance requirements
+ - Design constraints
+   - Standards Compliance
+ - Logical database requirement
+ - Software System attributes
+   - Reliability
+   - Availability
+   - Security
+   - Maintainability
+   - Portability
+ - Other requirements
+
+There is no hard and fast rule about how long a spec should be, but including all of that is just plain unnecessary for anything we'll be making in this class.
+Instead, let's distill some of the key ideas from above that our specs should touch on.
+
+***Disclaimer:*** Your teachers are by no means experts in writing specifications.
+The suggestions offered below are just that.
+The correct format will vary by project, so try to find what works best for you.
+
+- **Overview:** The story of your application, justification for why it needs to exist, and a broad overview of what it will accomplish.
```

Comment:
Were you planning to add examples of these, they are still pretty vague for most people I think.